src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java

```diff
@@ -520,6 +520,156 @@ public void xmlRouteBuilder_RestReportIdWorkloadInventory_IdParamGiven_ShouldCal
         camelContext.stop();
     }
 
+    @Test
+    public void xmlRouteBuilder_RestReportIdWorkloadInventory_IdParamGiven_ShouldCallFindByAnalysisIdAndReturnFilteredCsv() throws Exception {
+        //Given
+        Map<String, Object> variables = new HashMap<>();
+        Long one = 1L;
+        variables.put("id", one);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set(TestUtil.HEADER_RH_IDENTITY, TestUtil.getBase64RHIdentity());
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+        //When
+        camelContext.start();
+        TestUtil.startUsernameRoutes(camelContext);
+        camelContext.startRoute("to-sortBean");
+        camelContext.startRoute("to-workloadInventoryFilterBean");
+        camelContext.startRoute("filtered-workload-inventory-report-get-details-as-csv");
+        camelContext.startRoute("workload-inventory-report-model-to-csv");
+
+        ResponseEntity<String> response = restTemplate.exchange(camel_context + "report/{id}/workload-inventory/filtered-csv", HttpMethod.GET, entity, String.class, variables);
+
+        //Then
+        SortBean sortBean = new SortBean("id", false);
+        WorkloadInventoryFilterBean filterBean = new WorkloadInventoryFilterBean();
+
+        verify(workloadInventoryReportService).findByAnalysisOwnerAndAnalysisId("mrizzi@redhat.com", one, sortBean, filterBean);
```

Comment:
We could take advantage that we have the Spring context up and H2 and check that really the values filtered by Hibernate are the ones we expect.

Reply:
@jonathanvila as you can see in this line https://github.com/carlosthe19916/xavier-integration/blob/443b2364733bf2ffab182329e08454b07b4f12be/src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java#L611 and in the lines below it, there is an `analysisModel` and 3 `worloadInventoryReportModel` which are created in the DB. The final asserts of the method use the DB data